### PR TITLE
CircleCI to Github Migration

### PR DESCRIPTION
**Pull request by gokhangulbiz**

### CODE REVIEW

File: .github/workflows/extension.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          -u ${{ fromJson(secrets.AZURE_CREDENTIALS)['clientId']}} \
+          -p ${{ fromJson(secrets.AZURE_CREDENTIALS)['clientSecret']}} \
+          --tenant ${{ fromJson(secrets.AZURE_CREDENTIALS)['tenantId']}}
+    - name: Run extension tests
+      run: |-
+        mkdir ~/.ssh
+        cd ./azure
+        export EXTENSION_TEST=1
+        ./citus-bot.sh citusbot_extension_test_resource_group
+      shell: bash
```
Comment (gurkanindibay) [inline .github/workflows/extension.yml:39]: Could you add an emtpy line for all files?

**Comment by gurkanindibay** [comment]
Branch triggers are seperated by branch prefixes. Do the added pipelines are totally isolated so that other tests does not run when a push committed for one? @aykut-bozkurt do you have any idea?
Additionally, we need to see the results as well. Could you add links for successful execution for each pipeline?

**Review by gurkanindibay — CHANGES_REQUESTED** [review]

**Comment by gokhangulbiz** [comment]
> Branch triggers are seperated by branch prefixes. Are the added pipelines totally isolated so that other tests does not run when a push committed for one? @aykut-bozkurt do you have any idea?

Here is an example workflow run for tpch tests: https://github.com/citusdata/test-automation/actions/runs/6795655949 Based on this we can say they're totally isolated. 

> Additionally, we need to see the results as well. Could you add links for successful execution for each pipeline?

I believe workflow outputs include a link to the pushed repository, similar to what we have in CircleCI. https://github.com/citusdata/test-automation/actions/runs/6795655949/job/18474146284#step:7:1206

File: .github/workflows/cleanup.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            -u ${{ fromJson(secrets.AZURE_CREDENTIALS)['clientId']}} \
+            -p ${{ fromJson(secrets.AZURE_CREDENTIALS)['clientSecret']}} \
+            --tenant ${{ fromJson(secrets.AZURE_CREDENTIALS)['tenantId']}}
+      - name: Delete resource group
+        run: |-
+            mkdir ~/.ssh
+            cd ./azure
+            ./delete-resource-group-job.sh
+        shell: bash
+
```
Comment (gurkanindibay) [inline .github/workflows/cleanup.yml:37]: missing empty line for all files

**Comment by gurkanindibay** [comment]
Well done @gokhangulbiz . I like the style you design each pipeline. 
Can I see successful executions for these pipelines? Since they are designed to triggered by branch patterns, they are not triggered so I can not see the executions here. Additionally, we need to add trigger mechanism somewhere in the Readme to make the developers understand the trigger structure

**Comment by gokhangulbiz** [comment]
> Can I see successful executions for these pipelines?

Extension Tests : https://github.com/citusdata/test-automation/actions/runs/6860289670
JDBC Tests: https://github.com/citusdata/test-automation/actions/runs/6860283899
PGBench Tests: https://github.com/citusdata/test-automation/actions/runs/6860279431
Scale Tests: https://github.com/citusdata/test-automation/actions/runs/6860267307
TPCH Tests: https://github.com/citusdata/test-automation/actions/runs/6795655949

> Additionally, we need to add trigger mechanism somewhere in the Readme to make the developers understand the trigger structure

We have the similar trigger mechanism in GHA as documented in [readme.md](https://github.com/citusdata/test-automation#:~:text=There%20is%20a%20separate%20job%20for%20each%20test%20and%20you%20can%20run%20any%20combinations%20of%20them.%20To%20trigger%20a%20job%2C%20you%20should%20create%20a%20branch%20which%20has%20specific%20prefixes.) Also, I updated the documentation for minor naming changes in [731e4c1](https://github.com/citusdata/test-automation/pull/274/commits/731e4c1152d9ffaa7b2e6962a2d12f89358553aa)

**Review by aykut-bozkurt — APPROVED** [review]
I have checked `extension` job results. They look fine. I am not so sure about the other jobs' results even if they look fine at the first insight, it could be better to ask for people that had experience with them. Other than that, LGTM.

**Review by hanefi — APPROVED** [review]
I checked all workflow outputs, and all seemed to be running fine. We have many warnings that are not regressions.

Thanks for running all the workflows and sharing links in a comment for each of them.

**Pull request merged**